lib/landing_session.js

```diff
@@ -130,6 +132,10 @@ class LandingSession extends Session {
     return command;
   }
 
+  canAutomaticallyRebase(subjects) {
+    return subjects.every(line => !line.startsWith('squash!'));
```

Comment:
I suggest we don't check the commits, `fixup!` as a prefix is also valid, and if all commits are valid we still want to autorebase. The last ncu check will catch any commit not following the guidelines, so it should be safe to skip this check here

Reply:
Ahh, I see, it's skipping squash. Should be fine for a first version, but we want to handle squash as well

Reply:
> fixup! as a prefix is also valid,

Yep, that's why it is not here.

My main problem with allowing `squash!` here is that it will just throw every commit-message and commit-title together and the result is unlikely to be what the user wants. This is fine for manual landing (we can always change the message ourselves) but is not compatible with `commit-queue` as it will open an editor for commit-message editing.
Should we perhaps add another option for auto-landing where this check will be performed?

Reply:
Should we change the option to 
`--rebase=auto` -- skip squash!
`--rebase` || `--rebase=all` - allow squash, the default

?

Reply:
I think we should programmatically change squash with fixup before (or while) running rebase, squash is never what users want on nodejs/node. But let's land this as is and we can improve in a follow up, this is already way better than what we have today. Thank you for working on it!

Reply:
FWIW I don't think we should assume squash==fixup. I would be okay treating squash! commits as not landable via commit queue.

Reply:
Can you elaborate on why we shouldn't (for nodejs/node)? `squash!` will always be invalid there.

Reply:
I agree `squash!` would be invalid there. If it is used in a PR I would want commit queue to treat that as invalid and not second guess that it is to be treated as fixup.

Reply:
I suggested replacing squash with fixup because I want to minimize the cases where we leave the burden to collaborators (having to go to the terminal to land a PR is a very heavyweight process). If `commit-queue` doesn't handle squash it might increase the burden on collaborators. We could disallow squash as part of https://github.com/nodejs/node/pull/32417, but then we're shifting the burden to the author of the PR (which again, could be avoided with automation). IMO this is a case where absolute perfect correctness is less desirable than improved user experience. As a compromise, the linter could _warn_ the user that squashes will be converter into fixups (instead of asking users to open the terminal, checkout the code, rebase changing the commits, and force push).

Reply:
We should probably take this conversation outside of this PR 😄.

Reply:
I completely agree with @richardlau that assuming `squash! == fixup!` is not the right choice in general since they have a different meaning for both git and a user and the person writing `squash!` may want to combine/change the messages later.

@mmarchini I understand your point and looking at the PRs into Node.js people usually use `squash!` as a `fixup!` so it makes sense to just replace them but in a rare case where someone wants to change the message it's better to just ask them to do so manually and either change to `fixup!` or squash everything the way they want and force-push to the branch instead of second-guessing in the automation. (should we perhaps add a separate label for `fixup-everrything-to-first-commit`?)
IMO changing something in your branch is not that much of a problem and is done easily. Landing PRs to master, on the other hand, is what introduces the most burden.

Reply:
> IMO changing something in your branch is not that much of a problem and is done easily. Landing PRs to master, on the other hand, is what introduces the most burden.

That's a fair assumption once you get used to it, but not all new contributors are familiar enough with git for this task to be considered easy. Even existing collaborators might find it hard or a burden (note that burden !== easy in this case, something can be easy and still impose a burden barrier).

Maybe we should add a `manual-land` label that prevents the commit queue from attempting to land, in cases where the author or collaborator landing wants to change something before landing.

I still disagree on not assuming `squash == fixup` but that can be discussed outside this PR.

Reply:
fyi I didn't land this yet because I want to test against a PR or two first, but none of the discussion above is blocking.

Reply:
Sure, I still need to fix the nits =)

> Even existing collaborators might find it hard or a burden (note that burden !== easy in this case, something can be easy and still impose a burden barrier).

Yeah, that's what I thought when suggested the `fixup-everything-to-first-commit`. It would be even better to be able to just specify a commit message in a PR-comment, though I think it's currently not favored due to notifications.

> Maybe we should add a manual-land label that prevents the commit queue from attempting to land, in cases where the author or collaborator landing wants to change something before landing.

+1